test/unit/command_test.rb

```diff
@@ -66,5 +66,54 @@ class CommandTest < ActiveSupport::TestCase
       end
       assert_equal '/etc/passwd: Permission denied', error.message
     end
+
+    test 'sets code and message correctly on success' do
+      command = Command.new('true', chdir: '.')
+      assert_nil command.code
+      command.run
+      refute_predicate command, :running?
+      assert_predicate command.code, :zero?
+      assert_equal 'terminated successfully', command.termination_status
+    end
+
+    test 'sets code and message correctly on error' do
+      command = Command.new('false', chdir: '.')
+      assert_nil command.code
+      command.run
+      refute_predicate command, :running?
+      assert_predicate command.code, :nonzero?
+      assert_equal 'terminated with exit status 1', command.termination_status
+    end
+
+    test 'handles externally signalled commands correctly' do
+      command = Command.new('sleep 10', chdir: '.')
+      t = Thread.new do
+        signalled = false
+        20.times do
```

Comment:
why do we need to repeat this?

Reply:
I want to avoid the thread spinning forever, this limits its life to 2 seconds (20x0.1).

Reply:
Why do we need the loop in the first place? Would this not work if we only signal it once?

Reply:
This will only signal the command once - it breaks immediately after. I added this to protect myself when breaking something else in the code - mistyping a path or causing a loop etc. Without some kind of circuit breaker `thread.join` will never return in cases where the command is not started.